front/poke/temporal/activities.ts

```diff
@@ -40,3 +85,424 @@ export async function scrubDataSourceActivity({
+            }
+            await ConversationParticipant.destroy({
+              where: { conversationId: c.id },
+              transaction: t,
+            });
+            await c.destroy({ transaction: t });
+          })();
+        })
+      );
+    }
+  });
+}
+
+export async function deleteAgentsActivity({
+  workspaceId,
+}: {
+  workspaceId: string;
+}) {
+  const auth = await Authenticator.internalBuilderForWorkspace(workspaceId);
+  const workspace = auth.workspace();
+
+  if (!workspace) {
+    throw new Error("Could not find the workspace.");
+  }
+
+  const agents = await AgentConfiguration.findAll({
+    where: {
+      workspaceId: workspace.id,
+    },
+  });
+  await front_sequelize.transaction(async (t) => {
+    for (const agent of agents) {
+      if (agent.generationConfigurationId) {
+        await AgentGenerationConfiguration.destroy({
+          where: {
+            id: agent.generationConfigurationId,
+          },
+          transaction: t,
+        });
+      } else if (agent.retrievalConfigurationId) {
+        await AgentDataSourceConfiguration.destroy({
+          where: {
+            retrievalConfigurationId: agent.retrievalConfigurationId,
+          },
+          transaction: t,
+        });
+        await AgentRetrievalConfiguration.destroy({
+          where: {
+            id: agent.retrievalConfigurationId,
+          },
+          transaction: t,
+        });
+      } else if (agent.dustAppRunConfigurationId) {
+        await AgentDustAppRunAction.destroy({
+          where: {
+            dustAppRunConfigurationId: agent.dustAppRunConfigurationId,
+          },
+          transaction: t,
+        });
+        await AgentDustAppRunConfiguration.destroy({
+          where: {
+            id: agent.dustAppRunConfigurationId,
+          },
+          transaction: t,
+        });
+      }
+      await AgentUserRelation.destroy({
+        where: {
+          agentConfigurationId: agent.id,
+        },
+        transaction: t,
+      });
+      await agent.destroy({ transaction: t });
+    }
+    await GlobalAgentSettings.destroy({
+      where: {
+        workspaceId: workspace.id,
+      },
+      transaction: t,
+    });
+  });
+}
+
+export async function deleteAppsActivity({
+  workspaceId,
+}: {
+  workspaceId: string;
+}) {
+  const coreAPI = new CoreAPI(logger);
+  const auth = await Authenticator.internalBuilderForWorkspace(workspaceId);
+  const workspace = auth.workspace();
+
+  if (!workspace) {
+    throw new Error("Could not find the workspace.");
+  }
+
+  const apps = await App.findAll({
+    where: { workspaceId: workspace.id },
+  });
+
+  await front_sequelize.transaction(async (t) => {
+    for (const app of apps) {
+      const res = await coreAPI.deleteProject({
+        projectId: app.dustAPIProjectId,
+      });
+      if (res.isErr()) {
+        throw new Error(
+          `Error deleting Project from Core: ${res.error.message}`
+        );
+      }
+      await Run.destroy({
```

Comment:
Technically you might be deleting another's workspace run here. But I think that's fine :+1:

Reply:
That's if I delete a workspace who published apps that were used by another workspace in a agent?

Reply:
Yes